File: php/UsageExample.php
<?php
	
	require_once("AppChains.php");

	function printReport($chainsResult)
    {
        if ($chainsResult->isSucceeded())
            echo "Request has succeeded\n";
        else
            echo "Request has failed\n";

        foreach ($chainsResult->getResults() as $r) {
            $type = $r->getValue()->getType();

            switch ($type) {
                case ResultType::TEXT:
                    $v = $r->getValue();
                    echo sprintf("-> text result type %s = %s\n", $r->getName(), $v->getData());
                    break;
                case ResultType::FILE:
                    $v = $r->getValue();
                    echo sprintf(" -> file result type %s = %s\n", $r->getName(), $v->getUrl());
                    $v->saveTo("/tmp");
                    break;
            }
        }
    }

    $chains = new AppChains("<your token goes here>", "api.sequencing.com");
	
	echo "Beacons\n";
	/*$beaconResult = $chains->getPublicBeacon(1, 2, "A");
	print $beaconResult;*/

	echo "Chains\n\n";
	$chainsRawResult = $chains->getRawReport("StartApp", "Chain9", "227680");
	print_r($chainsRawResult);
		
	$chainsResult = $chains->getReport("StartApp", "Chain90", "227680");

    /**
     * @param $chainsResult
     */
    printReport($chainsResult);

    $chainsBatchResult = $chains->getBatchReport("StartAppBatch", array("Chain91" => "227680", "Chain88" => "227680"));

    foreach ($chainsBatchResult as $key => $value){
        echo "-> Chain ID:";
        echo $key;
        echo "\n";
        printReport($value);
    }


	
?>

File: perl/UsageExample.pl
#!/usr/bin/perl -w

use strict;
use warnings;
use AppChains;
use Data::Dumper;

my $chains = AppChains->new("<your token goes here>", "api.sequencing.com");

# single result retrieval
my $chainsResult = $chains->getReport("StartApp", "Chain9", "227680");

# batch result retrieval
my %chainsSpec = ("Chain9" => "227680", "Chain88" => "227680");
my $out = $chains->getBatchReport("StartAppBatch", \%chainsSpec);

foreach (keys %$out)
{
	my $cr = $out->{$_};
	print "-> Chain ID: " . $_ . "\n";
	handleAppChainResult($cr);
	print "\n\n";
}


sub handleAppChainResult
{
	my ($chainsResult) = @_;
	
	if ($chainsResult->isSucceeded()) {
		print "\tRequest has succeeded\n";
	} else {
		print "\tRequest has failed\n";
	}

	foreach (@{$chainsResult->getResults()})
	{
		my $type = $_->getValue()->getType();
	
		if ($type == ResultType->TEXT)
		{
			my $v = $_->getValue();
			print sprintf("\t-> text result type %s = %s\n", $_->getName(), $v->getData());
		}
		elsif ($type == ResultType->FILE)
		{
			my $v = $_->getValue();
			print sprintf("\t-> file result type %s = %s\n", $_->getName(), $v->getUrl());
			$v->saveTo("/tmp");
		}
	}
}
